src/main/appBadge.ts

```diff
@@ -25,7 +26,18 @@ export function setBadgeCount(count: number) {
     switch (process.platform) {
         case "linux":
             if (count === -1) count = 0;
-            app.setBadgeCount(count);
+            // app.setBadgeCount(count);
+            
+            function emitDBusBadge(count: number, visible: boolean) {
+                const badgeCountCommand = `gdbus emit --session --object-path / --signal com.canonical.Unity.LauncherEntry.Update "application://vesktop.desktop" "{'count': <int64 ${count}>, 'count-visible': <${visible}>}"`;
```

Comment:
since count comes from renderer via ipc, it is an untrusted value. embedding it into an `exec` command is a veeery bad idea. what if the browser passes something like `"$(rm -rf /*)"`?

you absolutely need to validate the value to avoid sandbox escapes, e.g. via `typeof count === "number"`

Reply:
also raw exec with string is not very robust, prefer calls that take an arg array, e.g. `execFile("gdbus", ["emit", ...])`. that's also more performant since it won't spawn an extra shell